samples/snippets/csharp/VS_Snippets_CLR/conceptual.globalization/cs/sortkey1.cs

```diff
@@ -6,8 +6,13 @@
 
 public class SortKeyComparer : IComparer<String>
 {
-    public int Compare(string str1, string str2)
+    public int Compare(string? str1, string? str2)
     {
+        if (str1 == null && str2 == null) return 0;
+
+        if (str1 == null) return -1;
+        if (str2 == null) return 1;
+
         SortKey sk1, sk2;
```

Comment:
I'm not convinced this is better, but here's a pattern matching equivalent (which should extend to the `return` statement, but GH UI won't allow a suggestion on that span):

```suggestion
        return (str1, str2) switch
        {
            (null, null) => 0,
            (null, _) => -1,
            (_, null) => 1,
            (var s1, var s2) => SortKey.Compare(
                CultureInfo.CurrentCulture.CompareInfo.GetSortKey(s1),
                CultureInfo.CurrentCulture.CompareInfo.GetSortKey(s1)),
        }
```